src/Sylius/Component/Core/Currency/ChannelAwareCurrencyProvider.php

```diff
@@ -48,4 +49,18 @@ public function getAvailableCurrencies()
             return $currency->isEnabled();
         });
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getBaseCurrency()
+    {
+        $currentChannel =  $this->channelContext->getChannel();
+
+        foreach ($currentChannel->getCurrencies() as $currency) {
+            if ($currency->isBase()) return $currency;
+        }
+
+        return;
```

Comment:
Pawel said that it should be mandatory to have a base currency, I'm not sure if we should through an exception here or not... If you do want to return null then it needs to be added to the Interface as a return type `|null`